### ISSUE TITLE
IRCv3 Tags contain empty values

### ISSUE THREAD

**Issue by eaceaser** [justintv/Twitch-API#393]
Here is an example:

```
@color=#0000FF;display-name=eaceaser;emotes=;subscriber=0;turbo=0;user-type=;user_type= :eaceaser!eaceaser@eaceaser.tmi.twitch.tv PRIVMSG #eaceaser :test
```

See the `user_type` keys (of which there are duplicates, too). This isn't a huge issue to work around, but this does violate the IRCv3 spec

```
<tag>           ::= <key> ['=' <escaped value>]
```

Which will be an issue for others developing clients against this implementation.

**Comment by TournyMasterBotCurse** [comment]
user_type is a deprecated item that will be removed soon, use user-type

**Comment by kaniini** [comment]
It does not violate the IRCv3 spec, a value may be 0 or more bytes.  `key=` is different than `key`.

**Comment by KlassenKonstantin** [comment]
Now I have the problem that "display-name" sometimes is empty, which loocks like this:

@color=#FF69B4;display-name=;emotes=41:0-7;subscriber=0;turbo=0;user-type= :xxx!xxx@xxx.tmi.twitch.tv PRIVMSG #clgdoublelift :Kreygasm

I could use xxx but it does not necessarily match the case. Any idea why the display-name could be empty?

**Comment by bashtech** [comment]
If display name is empty it was never set so assume the default of
capitalizing the first letter.

Can you provide examples of users that have to v3 display name but actually
have a display name set?

**Comment by Izlsnizzt** [comment]
Per https://github.com/justintv/Twitch-API/blob/master/IRC.md#privmsg

display-name is the user's display name, escaped as described as described
> in the IRCv3 spec <http://ircv3.net/specs/core/message-tags-3.2.html>.
>
>    - Empty if it's never been set.
>
>

**Comment by KlassenKonstantin** [comment]
> Can you provide examples of users that have to v3 display name but actually
have a display name set?

Is there a typo D:? Could you rephrase that?

**Comment by bashtech** [comment]
A user that has a display name set to something other then starting with a
capital letter according to https://api.twitch.tv/kraken/users/:user but
returns an empty tag in IRC.

**Comment by KlassenKonstantin** [comment]
Got it. I don't think it happens. It was my bad. I thought it would be all lowercase if not set. Thanks a lot!
